ext/semian/sysv_semaphores.c

```diff
@@ -130,22 +139,29 @@ get_semaphore(int key)
   return semget(key, SI_NUM_SEMAPHORES, 0);
 }
 
+struct acquire_semaphore_args {
+  semian_resource_t *res;
+  int *wait_time_ms;
+};
+
 void *
-acquire_semaphore_without_gvl(void *p)
+acquire_semaphore_without_gvl(semian_resource_t *res, int *wait_time_ms)
```

Comment:
Is the main purpose of this to figure out how long we are stuck waiting on a syscall without the GVL?